draft-ietf-anima-brski-prm.md

```diff
@@ -336,84 +320,84 @@ Solution examples based on existing technology are provided with the focus on ex
-              .         | (e.g., PKI Certificate |  .
-              .         |       Authority)       |  .
-              .         +------------------------+  .
-              .......................................
-                       "Domain" components
+                         +---------------------------+
+    +---- Drop Ship -----| Vendor Service            |
+    |                    +---------------+-----------+
+    |                    | M anufacturer |           |
+    |                    | A uthorized   | Ownership |
+    |                    | S igning      | Tracker   |
+    |                    | A uthority    |           |
+    |                    +---------------+-----------+
+    |                                         ^
+    |                                         | BRSKI-
+    |                                         | MASA
+    |          ...............................|.........
+    V          .                              v        .
++--------+     .  +------------+        +-----------+  .
+|        |     .  |            |        |           |  .
+| Pledge | BRSKI- | Registrar- | BRSKI- | Domain    |  .
+|        |  PRM   | Agent      |  EST   | Registrar |  .
+|        |<------>|            |<------>| (PKI RA)  |  .
+|        |     .  |     LDevID |        |           |  .
+|        |     .  +------------+        +-----+-----+  .
+| IDevID |     .                              |        .
+|        |     .           +------------------+-----+  .
++--------+     .           | Key Infrastructure     |  .
+               .           | (e.g., PKI Certificate |  .
+               .           |        Authority)      |  .
+               .           +------------------------+  .
+               .........................................
+                         "Domain" Components
 ~~~~
-{: #uc2figure title='BRSKI-PRM Architecture overview using registrar-agent' artwork-align="left"}
+{: #uc2figure title='BRSKI-PRM architecture overview using registrar-agent' artwork-align="left"}
 
 The following list describes the components in a (customer) site domain:
 
 * Pledge: The pledge is expected to respond with the necessary data objects for bootstrapping to the registrar-agent.
   The protocol used between the pledge and the registrar-agent is assumed to be HTTP in the context of this document.
-  Other protocols may be used like CoAP, Bluetooth, or NFC, but are out of scope of this document.
-  A pledge acting as a server during bootstrapping leads to some differences to BRSKI:
+  Other protocols such as CoAP, NFC, or Bluetooth may be used, but are out of scope of this document.
+  A pledge acting as a server during bootstrapping leads to some differences for BRSKI:
 
   * Discovery of the pledge by the registrar-agent must be possible.
 
   * As the registrar-agent must be able to request data objects for bootstrapping of the pledge, the pledge must offer corresponding endpoints.
```

Comment:
"must" or "MUST"?
Also, this falls back to a vague requirement discussion. Give the normative statements here.

Reply:
We now have the "MUST", but the statement does not really say what exactly MUST be done.

This is where my problem of the current structure of Section 5 comes in. For clear implementability, we need statements such as "The pledge MUST implement all endpoints listed in Table 1." I would collect all such statements for a specific component that needs to be implemented in its own section.

Currently Section 5 gives a lot but vague information in this bullet list under Figure 1 and then gives additional information in the behavior subsections. This makes it hard for an implementer find all relevant statements for their specific component.

Reply:
Okay, lets address this in the restructuring of section 5

Reply:
To prepare the restructuring from a normative statement point of view, I changed the sentence to 

_As the registrar-agent MUST be able to request data objects for bootstrapping of the pledge, the pledge MUST offer corresponding endpoints as defined in section 5.3._